src/librustc_typeck/check/_match.rs

```diff
@@ -963,6 +937,46 @@ https://doc.rust-lang.org/reference/types.html#trait-objects");
             self.check_pat_walk(&field.pat, field_ty, def_bm, true);
         }
 
+        if inexistent_fields.len() > 0 {
+            let field_names = if inexistent_fields.len() == 1 {
+                format!("a field named `{}`", inexistent_fields[0].1)
+            } else {
+                format!("fields named {}",
+                        inexistent_fields.iter()
+                            .map(|(_, name)| format!("`{}`", name))
+                            .collect::<Vec<String>>()
+                            .join(", "))
+            };
+            let spans = inexistent_fields.iter().map(|(span, _)| *span).collect::<Vec<_>>();
+            let mut err = struct_span_err!(tcx.sess,
+                                           spans,
+                                           E0026,
+                                           "{} `{}` does not have {}",
+                                           kind_name,
+                                           tcx.item_path_str(variant.did),
+                                           field_names);
+            for (span, name) in &inexistent_fields {
+                err.span_label(*span,
+                               format!("{} `{}` does not have field `{}`",
```

Comment:
I think the labels could just be `unknown field` instead of repeating all information

Reply:
or, since these are usually very compact, do something similar to use imports:

```
warning: unused imports: `u32`, `usize`
 --> src/main.rs:1:11
  |
1 | use std::{usize, u32};
  |           ^^^^^  ^^^
```

Reply:
[Changed](https://github.com/rust-lang/rust/pull/49160/files#diff-fd9ccd843c9940dcc501010d9b603807).